shell.nix

```diff
@@ -1,8 +1,17 @@
-# dfu-programmer doesn't have darwin on it's list of supported platforms
-{ pkgs ? import <nixpkgs> { config = { allowUnsupportedSystem = true; }; }
-, avr ? true, arm ? true, teensy ? true }:
+# Compatibility note:
+# This nix expression is designed to work on systems tracking the latest stable channel or unstable.
+# (Currently 19.03)
+{ pkgs ? import <nixpkgs> {}, avr ? true, arm ? true, teensy ? true }:
```

Comment:
Removed `config.allowUnsupportedSystem` here since `dfu-programmer` [now supports macOS officially](https://github.com/NixOS/nixpkgs-channels/blob/nixpkgs-19.03-darwin/pkgs/development/tools/misc/dfu-programmer/default.nix) in `nixpkgs-19.03-darwin` (and unstable of course).